runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRLogEntry.java

```diff
@@ -0,0 +1,129 @@
+package org.corfudb.protocols.logprotocol;
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ * A log entry structure which contains a collection of multiSMRentries,
+ * each one contains a list of updates for one object.
+ */
+@SuppressWarnings("checkstyle:abbreviation")
+@ToString
+@Slf4j
+public class SMRLogEntry extends LogEntry {
```

Comment:
Is it possible to remove LogEntry, or just use LogData. I think we should just use one, but not both.

Reply:
Actually LogEntry is useful when there are different types of entry inside LogData, such as later we will have SMRGarbageEntry for new GC.